### ISSUE TITLE
bnum overrides features chosen for num-integer and num-traits

### ISSUE THREAD

**Issue by lrubasze** [isaacholt100/bnum#10]
I need to use num-traits or num-integer without default features (`no_std` in fact), eg.

`Cargo.toml`:
```
...
[dependencies]
num-traits = { version = "0.2.15", default-features = false }
num-integer = { version = "0.1.45", default-features = false }
bnum = { version = "0.3", default-features = false, features = ["numtraits"] }
...
```
In such configuration `bnum` overrides my settings and sets num-integer and num-traits to default (with `std`).

**Comment by lrubasze** [comment]
It looks like other dependencies: `serde`, serde-big-array` and `rand` might also override the features chosen.

**Comment by isaacholt100** [comment]
Fixed. All dependencies now have `default-features = false`, so you should be able to use the new release (v0.4.0) with `no_std`.

**Comment by lrubasze** [comment]
thanks for prompt fix and release. really appreciate it 👍 
btw. 
discovered that there is some issue here: https://docs.rs/crate/bnum/latest

![image](https://user-images.githubusercontent.com/117115317/210212580-4e3b61d7-62aa-495c-a56b-d72419c46586.png)

no idea, what could be the root cause.

**Comment by isaacholt100** [comment]
No worries at all, and thanks for pointing this out, I hadn't noticed this.

**Comment by isaacholt100** [comment]
Thanks again for mentioning the docs.rs build error, I have now fixed this issue, it was because v0.4.0 didn't compile with the latest version of nightly. v0.5.0 has now been published which fixes this.
